config/cmd/configure.go

```diff
@@ -41,6 +41,8 @@ type DefaultsData struct {
 	InitImagePrefix  string
 	ImagePrefix      string
 	BuildImagePrefix string
+	PodTimeout       string
```

Comment:
The timeout should not be specified in defaults, but should be derived from the test timeout. I think it does make sense to specify "KillAfter" as some sort of global default.

Reply:
I think it might be better to make PodTimeout a structure in the LoadTest configuration, with fields like GracePeriodSeconds and KillAfterSeconds. The first one would be added to the test timeout to create the pod timeout. In any case, we should consider the interaction between the test timeout and the pod timeout – do we want a race between the pod and the test timing out? would that affect the end state? We should give it some thought.

Reply:
Right, I create this PR to collect thoughts on how much flexibility we want on Podtimeout and Killafter.  I agree that Killafter is somewhat default that most user doesn't have to touch, it is  just to make sure that the timeout is executed. I can see the  Podtimeout time to be set through LoadTest configuration files, since it has the similar effect as TimeoutSeconds.

The Podtimeout count start the moment when the run container starts, it protects the case that the pod is successfully get to run container but the process within run container went wrong. 

Timeoutseconds tries to guard the total amount of time starting from the LoadTest being scheduled, but the only thing it does is to mark the LoadTest as TimeoutErrored and send asynchronous RPC to workers, there is no guarantee the workers will be stopped.  (1) With scenarios like the building process went wrong (not using prebuild image), if initial containers take longer than Timeoutseconds, the LoadTest will be marked as TimeoutErrored, but the workers will ignore the RPC. The behavior of Timeoutseconds is similar to not any control. (2) If the initial process went through okay, the process in run containers goes wrong, TimououtSeconds may be able stop the workers, which is also similar to what Podtimeout could achieve for sure.

I am thinking of using Podtimeout to replace Timeoutseconds, and having driver and workers have same PodtimeoutSeconds, as Ready container timeout was set as Timeoutseconds, we may want to set it with PodtimeoutSeconds to give enough time for workers to be up and running. 

It should work like this:
With prebuilt image 
(1) Workers up and running okay --> Driver is up --> Driver Timeout --> LoadTest Marked as Timeout Error --> Workers timeout on their own
(2) Workers up and running okay --> Driver is up --> Worker Timeout --> LoadTest Marked as Timeout Error --> Driver timeout 
(3) Workers could not run (stuck before get to run container)--> Driver Ready Container Timeout -->  LoadTest Marked as Timeout Error, in this case the worker might stuck waiting for TTL seconds to delete the pods all together, which is what we have with  Timeoutseconds.

Just my thoughts, we could discuss further in the meeting. :)